### PR TITLE
Fixes for Python 3.12

### PR DESCRIPTION
**Pull request by nannerpusser**
A second PR attempt after failing with the last. This time diffs are included, very minor changes in syntax and proper escaping. Also fixed deprecated Mutables class from collections.	

modified:   README.md
modified:   scripts/parse-rc.py
modified:   scripts/update-rc.py
modified:   scripts/util.py

### CODE REVIEW

**Comment by gto76** [comment]
Thank you very much for spotting the outdated import in the python script.

However, your pull request is, to put it frankly, one big mess.
When you're fixing such an important issue (that only requires two line changes) your pull request should contain one commit that only fixes that single issue.

That is why I updated the script myself and closed your pull request.

**Comment by nannerpusser** [comment]
> Thank you very much for spotting the outdated import in the python script.
> 
> However, your pull request is, to put it frankly, one big mess. When you're fixing such an important issue (that only requires two line changes) your pull request should contain one commit that only fixes that single issue.
> 
> That is why I updated the script myself and closed your pull request.

Oh, well it wasn't just the import, but the string type in the regex functions has to be defined now rather than defaulting to ``r``. Kind of a dumb, pedantic change, but in a terminal it throws too many warnings and exceptions now and I had never contributed to an actual real repo (or any code for that matter) so I thought I'd get my dick out and swing it around  a little. Little in both senses of the word, pun intended.

Tiny change, first time submitting a PR, first contribution to anyone's code. Probably should have RTFM on PRs before winging it, but hey, all good. Also, please ignore all the ASCII character corrections I had to do because I forgot I was on WSL and pulled that file across file systems and Windows gonna be Windows. Amateur Hour in here, I apologize again.

**Comment by gto76** [comment]
I forgot about the `r`s, thanks for that too.

However those changes were also confusing, since you put `r` in font of all strings containing regexes, except for one.
When put in context with other sloppy changes, like the unintentional ASCII, it just left me confused.
I don't know what it is, but you just give an AI vibe with how confident you sound, yet you're just a bit off on most things.

Like for instance here:
> ... the string type in the regex functions has to be defined now rather than defaulting to `r`.

Python does not have different string types. What `r` does is it prevents Python from changing escaped characters like `\n` or `\t` into actual new lines or tabs.
```python
>>> print('\tXXX')
	XXX
>>> print(r'\tXXX')
\tXXX
```

Regex strings that are passed to re functions usually contain a lot of escaped characters (`\d`, `\w`, ...) that are not recognized by Python. Python leaves these sequences intact.
```python
>>> print('\d')
\d
>>> print(r'\d')
\d
```

However it started throwing warnings recently when it encounters them.

If `r` is added in front of such strings than Python will not complain, however the code will break if strings contained any backreferences like `\\1`, because regex function will now receive `\\1` instead of `\1`; my point being that one has to be really careful when fixing this specific warnings.

Se even if you had nicely separated this `r` fixes into separate commit and pull request I would still be very suspicious because there was no explanation given as to why some strings got `r`ed and some not. Was this intentional? Was any automatic tool used to fix this warning? was AI involved? Was it tested?


Bottom line is you made an important contribution to a small project that I use daily and have it pinned on my profile page and because of that I don't want you to feel unwelcome. Also, a pull request, even in this state, is probably more useful than just a new issue. However it triggered my inner [Linus](https://github.com/corollari/linusrants?tab=readme-ov-file), for better or worse :)

**Comment by nannerpusser** [comment]
Oh hey, I get it and I even get how AI-ish it all looks. Gen AI code is almost as sloppy and ignorant as me with python, and that's hard to do. I gave up on it after OpenAI paywalled the GPT4 API (don't blame them, I'd have done it sooner and not allowed public 3.5 shenanigans without a credit card if I were hemorrhaging millions every month) and I wasn't in the circlejerk C-suite lounge where free enterprise dev keys and tokens were handed out.  

As for the regex strings, I absolutely didn't communicate that right at all. In my head I thought ``r`` meant ``raw`` and the totally improper and self-taught way I learned the tiny bit of python I know categorized ``raw`` as the type of regex string you parse into pieces. It of course is not a literal type since strings, as you said, have no types, but it *is* a type of regex string operation to my broken, wrong brain. I also missuse almost ever other python convention we have a name for as well. Tuple? Nope, brain never slotted that one right either, so its a "junior dict" or just small list when I read it. For that matter I learned the basics of python within Flask and really only dealt with JSON objects and never got used to dictionaries despite being one and the same. "JSON square bracket" and "JSON curly brace" are my brain names. So that's why I sound like a fuckin` goofy chat bot. Honestly. 

As for missing the one ``raw`` prefix, gonna chalk it up to being terrible at GitHub stuff, especially these interminable damn web IDEs, codespaces, and in-page editors I used that aren't ``eric`` or ``micro`` in my linux shell where things aren't so confusing.

**Comment by gto76** [comment]
I'm sorry. The combination of having to fix old code that I haven't touched in years with frustration about Python's new unnecessary warnings got to me for a moment.

I think the warnings are now gone and everything should be working as it was before. Please report if there are still any left, or if there is anything else wrong (I am still on older version of Python that doesn't report this warnings).

Thanks again.
